### PR TITLE
refactor: use BitArray for PTC votes instead of boolean[]

### PR DESCRIPTION
**Pull request by GrapeBaBa**
## Summary
- Replace `boolean[]` (512 elements, ~4KB per block on mainnet) with `BitArray` from `@chainsafe/ssz` (64 bytes per block on mainnet) for PTC vote storage in fork choice
- Use byte-level popcount (Brian Kernighan's algorithm) instead of `Array.filter()` for vote counting in `isPayloadTimely()`
- Aligns with how other bitvectors (`aggregationBits`, `syncnets`, etc.) are represented across the codebase

## Details
The PTC (Payload Timeliness Committee) tracks 512 validator votes per Gloas block. Previously stored as `boolean[]` (each vote = 8 bytes in V8), now stored as a `BitArray` backed by a 64-byte `Uint8Array` — an 8x memory reduction per tracked block.

The vote counting path (`isPayloadTimely()`) previously used `votes.filter(v => v).length` which allocates a new array. Now uses popcount on the underlying bytes directly.

No behavioral changes — all 99 existing fork-choice unit tests pass.

## Test plan
- [x] All 31 Gloas-specific unit tests pass (PTC vote tests exercise all edge cases)
- [x] All 99 fork-choice unit tests pass
- [x] Lint passes

*AI-assisted: refactoring identified and implemented with Claude*

### CODE REVIEW

**Review by wemeetagain — DISMISSED** [review]
lgtm

File: packages/fork-choice/src/protoArray/protoArray.ts [hunk 1]
```diff
@@ -20,6 +21,19 @@ import {
  */
 const PAYLOAD_TIMELY_THRESHOLD = Math.floor(PTC_SIZE / 2);
 
+/** Count set bits in a Uint8Array without allocation (Brian Kernighan's algorithm) */
+function bitCount(arr: Uint8Array): number {
```
Comment (nflaig) [inline packages/fork-choice/src/protoArray/protoArray.ts]: can we move this to packages/utils please

**Review by nflaig — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.41.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.41.0) :tada:
